components/brave_today/browser/brave_news_controller.cc

```diff
@@ -113,51 +118,166 @@ void BraveNewsController::GetPublishers(GetPublishersCallback callback) {
   publishers_controller_.GetOrFetchPublishers(std::move(callback));
 }
 
+void BraveNewsController::SubscribeToNewDirectFeed(
+    const GURL& feed_url,
+    SubscribeToNewDirectFeedCallback callback) {
+  // Verify the url points at a valid feed
+  VLOG(1) << "SubscribeToNewDirectFeed: " << feed_url.spec();
```

Comment:
We should filter the feed URL here to make sure that HTTP basic auth credentials don't end up in the logs. So maybe a `sanitizeForLogs(GURL)` function and removes these components of the URL if they exist.

Reply:
Even for `VLOG(1)` entries? 

If a user runs the browser via terminal with `--v=1 --enable-logging=stderr` (which is certainly not normal), then the regular logs contain a lot of private information - all extension IDs and all request URLs that all tabs make, including unsanitized urls:

```
9:0112/000042.141551:VERBOSE1:layout_shift_tracker.cc(354)] in "https://my:password@petemill.com/", 0x1500a0f208:LayoutNGBlockFlow P class='TAOwZ-Y1eSetukukGbysl _7nQSGpeYqLqUROW0UBrgu' moved from 486.000000,188.000000 600.000000x64.000000 to 486.000000,220.000000 600.000000x64.000000 (visible from 486,188 600x64 to 486,220 600x64)
[26925:259:0112/000042.141616:VERBOSE1:layout_shift_tracker.cc(354)] in "https://my:password@petemill.com/", 0x1500a0f3e0:LayoutNGBlockFlow P class='TAOwZ-Y1eSetukukGbysl _7nQSGpeYqLqUROW0UBrgu' moved from 486.000000,272.000000 600.000000x160.000000 to 486.000000,304.000000 600.000000x192.000000 (visible from 486,272 600x160 to 486,304 600x192)
```

Reply:
Well, both.

If you turn on this extra verbose debug logging (as opposed to regular logs) then chromium does credential urls if you navigate to them in a tab.

And perhaps at the deeper network levels there is logging. Hard to see as there're so many log lines when you turn this on!

Reply:
My concern was that we shouldn't add log statements that leak credentials because we have a history of asking users for these logs later (in the past, we have included OAuth tokens in similar messages for example).

That said if the raw feed URL is going to be present anyways at that log level, then I guess it's a moot point. It will be present in the logs regardless of what we do in our own code.

So if you think that it is definitely going to be present already, then there's no point in sanitizing the URLs. If you're not sure, there the safe thing would be to sanitize our `VLOG`s.